### PR TITLE
Reorganizing requirements

### PR DESCRIPTION
**Pull request by jarrodmillman**
## Description

This declutters the root directory (related to #2777) by creating a requirements directory.  I have a few things I want to look into before merging this, but I wanted to get some feedback on whether this might be worth pursuing.

### CODE REVIEW

**Comment by stefanv** [comment]
This looks good to me so far.

**Comment by soupault** [comment]
I think the approach is fine.

We'd need to adapt also https://github.com/scikit-image/scikit-image/blob/master/setup.py#L48-L49 (from what I remember). Probably, there are several more usages.

Regarding https://github.com/scikit-image/scikit-image/blob/master/DEPENDS.txt : I'd propose to drop it, and move any comments/descriptions into corresponding `requirement`-file. Seems like `pip` supports comment sections (https://github.com/pypa/pip/issues/564).

**Comment by stefanv** [comment]
While it would be convenient to drop the dependencies file, it is included in the documentation; we can think of auto-generating that version, but it's not a trivial task.

**Comment by soupault** [comment]
@stefanv From my perspective, it should be relatively easy to implement such function. Let's keep it for another PR though.

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,2 @@
-matplotlib>=1.3.1
-numpy>=1.11
-scipy>=0.17.0
-six>=1.7.3
-networkx>=1.8
-pillow>=2.1.0
-PyWavelets>=0.4.0
+-r requirements/default.txt
```
Comment (soupault) [inline requirements.txt]: Why keep this file?

**Comment by emmanuelle** [comment]
Thanks @jarrodmillman ! Tell us when you're ready and want us to merge the PR.

File: requirements.txt (hunk 1 above)
Comment (stefanv) [reply requirements.txt]: I think this is the default that everyone tries to install, so not a bad idea to have it.

**Comment by jarrodmillman** [comment]
This is ready to merge.

**Pull request merged**